skops/card/_parser.py

```diff
@@ -0,0 +1,353 @@
+"""Contains the PandocParser
+
+This class needs to know about the pandoc parse tree but should not have
+knowledge of any particular markup syntex; everything related to markup should
+be known by the mapping attribute.
+
+"""
+
+from __future__ import annotations
+
+import json
+import subprocess
+from pathlib import Path
+from tempfile import mkdtemp
+from typing import Any, Literal, Sequence
+
+import yaml  # type: ignore
+
+from skops.card import Card
+from skops.card._model_card import Section
+
+from ._markup import Markdown, PandocItem
+
+PANDOC_MIN_VERSION = (2, 19, 0)
+
+
+class PandocParser:
+    """Create model cards from files parsed through pandoc.
+
+    This class knows about the implementation details of the
+    :class:`~skops.card.Card` and generates it by initializing an empty class
+    and then calling its methods with the input provided by pandoc.
+
+    ``PandocParser`` does not know about any specific markup type, such as
+    markdown. Instead, it is initialized with a ``Mapping``, which is
+    responsible to convert pandoc input into the desired markup language.
+
+    After initializing this class, call
+    :meth:`~skops.card._parser.PandocParser.generate` to generate the resulting
+    :class:`~skops.card.Card` instance.
+
+    Parameters
+    ----------
+    source : str
+        The model card parsed using the ``pandoc -t json`` option.
+
+    markup_type : "markdown"
+        The type of markup that was used for the model card. Right now, only
+        ``"markdown"`` is supported.
+
+    """
+
+    def __init__(
+        self, source: str, markup_type: Literal["markdown"] = "markdown"
+    ) -> None:
+        self.source = source
+        if markup_type.lower() == "markdown":
+            self.mapping = Markdown()
+        else:
+            raise ValueError(f"Markup of type {markup_type} is not supported (yet)")
+
+    def _add_section(
+        self, section_name: str, card: Card, section_trace: list[str]
+    ) -> Section:
+        # Add a new section to the card, which can be a subsection, and return
+        # it.
+        section_name = "/".join(section_trace)
+        cur_section = card._add_single(section_name, "")
+        return cur_section
+
+    def _add_content(self, content: str, section: Section | None) -> None:
+        # Add content to the current section
+        if section is None:
+            # This may occur if the model card starts without a section. This is
+            # not illegal in markdown, but we don't handle it yet.
+            raise ValueError(
+                "Trying to add content but there is no current section, "
+                "this is probably a bug, please open an issue on GitHub"
```

Comment:
```suggestion
                "this is probably a bug, please open an issue on GitHub."
```

not sure why you like to miss the dots at the end of exception messages :D

Reply:
Not sure, probably because Python error messages also omit the period when it's only one sentence?